src/main/java/ru/sbt/mipt/oop/SmartHome.java

```diff
@@ -1,24 +1,36 @@
 package ru.sbt.mipt.oop;
 
+import ru.sbt.mipt.oop.smarthome.Action;
+import ru.sbt.mipt.oop.smarthome.Actionable;
+import ru.sbt.mipt.oop.smarthome.Room;
+
 import java.util.ArrayList;
 import java.util.Collection;
 
-public class SmartHome {
-    Collection<Room> rooms;
+public class SmartHome implements Actionable {
+    private final Collection<Room> rooms;
 
     public SmartHome() {
         rooms = new ArrayList<>();
     }
 
     public SmartHome(Collection<Room> rooms) {
-        this.rooms = rooms;
+        this.rooms = new ArrayList<>(rooms);
     }
 
-    public void addRoom(Room room) {
-        rooms.add(room);
+    public Room getRoomByName(String roomName) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%B7%D0%B0%D0%BA%D1%80%D0%BE%D0%B9%D1%82%D0%B5-%D0%B4%D0%BE%D1%81%D1%82%D1%83%D0%BF-%D0%BA-%D0%B2%D0%BD%D1%83%D1%82%D1%80%D0%B5%D0%BD%D0%BD%D0%B8%D0%BC-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D0%BC